### PR TITLE
Feature: Ethers signer

### PR DESCRIPTION
**Pull request by mikheevm**
This PR:
- Adds `SafeAppsSdkSigner` to `safe-apps-ethers-provider` package which allows calling smart contract functions just as you would in a dapp: `exampleContract.func()`

See in action: https://github.com/rmeissner/sapp-phenix/pull/2

### CODE REVIEW

**Comment by mikheevm** [comment]
https://gnosis-safe.io/cla/

**Comment by mikheevm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mikheevm** [comment]
recheckcla

**Comment by mikheevm** [comment]
recheckcla

**Comment by mikheevm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mikheevm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mikheevm** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by rmeissner — APPROVED** [review]

File: packages/safe-apps-ethers-provider/src/provider.ts [hunk 1]
```diff
@@ -28,9 +31,149 @@ function checkError(method: string, error: any): any {
...
+    message = error.body;
+  } else if (typeof error.responseText === 'string') {
+    message = error.responseText;
+  }
+  message = (message || '').toLowerCase();
+
+  const transaction = params.transaction || params.signedTransaction;
+
+  // "insufficient funds for gas * price + value + cost(data)"
+  if (message.match(/insufficient funds/)) {
```
[the remark this answers is not in the document]
Comment (mikheevm) [reply packages/safe-apps-ethers-provider/src/provider.ts:47]: yes I copied it from ethers.. I think this can be removed

**Pull request merged**